nebullvm/operations/root/black_box/operation.py

```diff
@@ -0,0 +1,314 @@
+    FetchDataFromLocal,
+)
+from nebullvm.operations.measures.measures import LatencyOriginalModelMeasure
+from nebullvm.operations.measures.utils import QUANTIZATION_METRIC_MAP
+from nebullvm.operations.optimizations.optimizers import (
+    PytorchOptimizer,
+    ONNXOptimizer,
+    TensorflowOptimizer,
+)
+from nebullvm.operations.root.black_box.utils import (
+    map_compilers_and_compressors,
+    check_input_data,
+    get_dl_framework,
+    extract_info_from_data,
+    is_huggingface_data,
+)
+from nebullvm.optional_modules.tensorflow import tensorflow as tf
+from nebullvm.optional_modules.torch import Module
+from nebullvm.tools.base import (
+    ModelCompiler,
+    DeepLearningFramework,
+    ModelParams,
+    OptimizationTime,
+)
+from nebullvm.tools.data import DataManager
+from nebullvm.tools.feedback_collector import FEEDBACK_COLLECTOR
+
+
+class BlackBoxModelOptimizationRootOp(Operation):
+    def __init__(self):
+        super().__init__()
+
+        self.model = None
+        self.data = None
+        self.optimal_model = None
+        self.conversion_op = None
+
+        self.fetch_model_op = FetchModelFromLocal()
+        self.fetch_data_op = FetchDataFromLocal()
+        self.orig_latency_measure_op = LatencyOriginalModelMeasure()
+        self.torch_conversion_op = PytorchConverter()
+        self.tensorflow_conversion_op = TensorflowConverter()
+        self.torch_optimization_op = PytorchOptimizer()
+        self.onnx_conversion_op = ONNXConverter()
+        self.onnx_optimization_op = ONNXOptimizer()
+        self.tensorflow_optimization_op = TensorflowOptimizer()
+
+    def _get_conversion_op(self, dl_framework: DeepLearningFramework):
+        if dl_framework == DeepLearningFramework.PYTORCH:
+            conversion_op = self.torch_conversion_op
+        elif dl_framework == DeepLearningFramework.TENSORFLOW:
+            conversion_op = self.tensorflow_conversion_op
+        else:
+            conversion_op = self.onnx_conversion_op
+
+        return conversion_op
+
+    def execute(
+        self,
+        model: Any,
+        input_data: Union[Iterable, Sequence],
+        metric_drop_ths: float = None,
+        metric: Union[str, Callable] = None,
+        optimization_time: str = "constrained",
+        dynamic_info: Dict = None,
+        config_file: str = None,
+        ignore_compilers: List[str] = None,
+        ignore_compressors: List[str] = None,
+        store_latencies: bool = False,
+        **kwargs,
+    ):
+        self.logger.info(
+            f"Running Black Box Nebullvm Optimization on {self.device.name}"
+        )
+
+        if self.fetch_model_op.get_model() is None:
+            self.fetch_model_op.execute(model)
+        if self.fetch_data_op.get_data() is None:
+            self.fetch_data_op.execute(input_data)
+
+        ignore_compilers = map_compilers_and_compressors(
+            ignore_compilers, ModelCompiler
+        )
+        # ignore_compressors = _map_compilers_and_compressors(
```

Comment:
compressors will surely be at the heart of nebullvm, in many apps we will develop. I would insert the logic for managing compressors anyway, even if we are considering to modify the pruning compressors.